src/promptflow/promptflow/_cli/_pf/_experiment.py

```diff
@@ -197,11 +226,41 @@ def show_experiment(args: argparse.Namespace):
     print(json.dumps(result._to_dict(), indent=4))
 
 
+def test_experiment(args: argparse.Namespace):
+    result = _get_pf_client()._experiments._test(args.name)
+    print(json.dumps(result._to_dict(), indent=4))
+
+
 def start_experiment(args: argparse.Namespace):
-    result = _get_pf_client()._experiments.start(args.name)
+    if args.name:
+        logger.debug(f"Starting a named experiment {args.name}.")
+        inputs = list_of_dict_to_dict(args.inputs)
+        if inputs:
+            logger.warning("The inputs of named experiment cannot be modified.")
+        client = _get_pf_client()
+        experiment = client._experiments.get(args.name)
+        result = client._experiments.start(experiment=experiment, stream=args.stream)
+    elif args.file:
+        from promptflow._sdk._load_functions import _load_experiment
+
+        logger.debug(f"Starting an anonymous experiment {args.file}.")
+        experiment = _load_experiment(source=args.file)
+        inputs = list_of_dict_to_dict(args.inputs)
+        result = _get_pf_client()._experiments.start(experiment=experiment, inputs=inputs, stream=args.stream)
+    else:
+        raise UserErrorException("To start an experiment, one of [name, file] must be specified.")
     print(json.dumps(result._to_dict(), indent=4))
 
 
 def stop_experiment(args: argparse.Namespace):
-    result = _get_pf_client()._experiments.stop(args.name)
+    client = _get_pf_client()
+    if args.name:
+        logger.debug(f"Stop a named experiment {args.name}.")
+        experiment = client._experiments.get(args.name)
+    elif args.file:
```

Comment:
So anonymous exp run does not have run record? This is a bit strange to stop an exp with its yaml file. I thought we will record the anonymous exp run but just give it a random exp name guid.

Reply:
+1 what's the target scenario of passing file here?

Reply:
This is a new added user experience in Evaluation Experience.
It is used to support users to directly execute from experience yaml. And it also supports future feature that users can define experiences through Python code.